src/main/java/fi/otavanopisto/kuntaapi/server/index/OrganizationSearcher.java

```diff
@@ -0,0 +1,98 @@
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.elasticsearch.action.search.SearchRequestBuilder;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHitField;
+
+import fi.otavanopisto.kuntaapi.server.id.OrganizationId;
+import fi.otavanopisto.kuntaapi.server.integrations.KuntaApiConsts;
+
+@ApplicationScoped
+public class OrganizationSearcher extends AbstractSearcher {
+  
+  private static final String TYPE = "organization";
+  private static final String ORGANIZATION_ID_FIELD = "organizationId";
+  private static final String BUSINESS_NAME_UT_FIELD = "businessNameUT";
+  private static final String BUSINESS_CODE_FIELD = "businessCode";
+  
+  @Inject
+  private IndexReader indexReader;
+
+  public SearchResult<OrganizationId> searchOrganizations(String queryString, String businessCode, String businessName, Long firstResult, Long maxResults) {
+    BoolQueryBuilder query = boolQuery().
+      must(queryStringQuery(queryString));
+    
+    if (businessCode != null) {
+      query.must(termQuery(BUSINESS_CODE_FIELD, businessCode));
+    }
+    
+    if (businessName != null) {
+      query.must(termQuery(BUSINESS_NAME_UT_FIELD, businessName));
+    }
+    
+    return searchOrganizations(query, firstResult, maxResults);
+  }
+   
+  public SearchResult<OrganizationId> searchOrganizationsByBusinessCode(String businessCode, Long firstResult, Long maxResults) {
+    return searchOrganizations(termQuery(BUSINESS_CODE_FIELD, businessCode), firstResult, maxResults);
+  }
+  
+  public SearchResult<OrganizationId> searchOrganizationsByBusinessName(String businessName, Long firstResult, Long maxResults) {
+    return searchOrganizations(termQuery(BUSINESS_NAME_UT_FIELD, businessName), firstResult, maxResults);
+  }
+  
+  public SearchResult<OrganizationId> searchOrganizationsByBusinessCodeAndBusinessName(String businessCode, String businessName, Long firstResult, Long maxResults) {
+    BoolQueryBuilder query = boolQuery()
+      .must(termQuery(BUSINESS_CODE_FIELD, businessCode))
+      .must(termQuery(BUSINESS_NAME_UT_FIELD, businessName));
+    
+    return searchOrganizations(query, firstResult, maxResults);
+  }
+  
+  private SearchResult<OrganizationId> searchOrganizations(QueryBuilder queryBuilder, Long firstResult, Long maxResults) {
+    SearchRequestBuilder requestBuilder = indexReader
+        .requestBuilder(TYPE)
+        .storedFields(ORGANIZATION_ID_FIELD)
+        .setQuery(queryBuilder);
+    
+    if (firstResult != null) {
+      requestBuilder.setFrom(firstResult.intValue());
+    }
+    
+    if (maxResults != null) {
+      requestBuilder.setSize(maxResults.intValue());
+    }
+      
+    return new SearchResult<>(getOrganizationIds(indexReader.search(requestBuilder)));
+  }
+  
+  private List<OrganizationId> getOrganizationIds(SearchHit[] hits) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "getOrganizationIds" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)